eoscompanion/v2_routes.py

```diff
@@ -0,0 +1,551 @@
+# /eoscompanion/v2_routes.py
+#
+# Copyright (C) 2017, 2018 Endless Mobile, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+# All rights reserved.
+'''V2 route definitions for eos-companion-app-service.'''
+
+from collections import defaultdict
+import logging
+import os
+
+
+from gi.repository import (
+    ContentFeed,
+    EosCompanionAppService,
+    GLib
+)
+
+from .applications_query import (
+    application_listing_from_app_info
+)
+from .format import (
+    format_app_icon_uri,
+    format_content_data_uri,
+    format_thumbnail_uri,
+    parse_uri_path_basename
+)
+from .functional import all_asynchronous_function_calls_closure
+from .middlewares import (
+    add_content_db_conn,
+    apply_version_to_all_routes,
+    record_metric,
+    require_query_string_param
+)
+from .responses import (
+    json_response,
+    serialize_error_as_json_object
+)
+from .v1_routes import (
+    companion_app_server_application_colors_route,
+    companion_app_server_application_icon_route,
+    companion_app_server_content_data_route,
+    companion_app_server_content_metadata_route,
+    companion_app_server_device_authenticate_route,
+    companion_app_server_license_route,
+    companion_app_server_list_application_content_for_tags_route,
+    companion_app_server_list_application_sets_route,
+    companion_app_server_list_applications_route,
+    companion_app_server_resource_route,
+    companion_app_server_search_content_route
+)
+
+
+def desktop_id_to_app_id(desktop_id):
```

Comment:
Most of this code is identical to the `v1` file. We should not have two copies of the code that can be shared.

Reply:
Oh, I see that this in a later commit this code is actually moved over from the v1 one. Are there any changes at all between the v1 code and the v2 code? If no, this is probably fine (but then why are we doing the API bump again!?), but if so, I think the best implementation would be to move the code over as-is in a single commit, then apply the v2-specific changes on top in a separate commit.

Reply:
Probably the most relevant change is that we introduced the `content-type` field a little bit after we branched. So there are some deployed companion app service versions out there that don't include `content-type` on each item in the feed response and the feed in the app will depend on that.

I remember that on our call two weeks ago we discussed that not having the feed at all in /v1 (the currently released companion app has no concept of the feed) and putting the whole thing behind /v2 helps with any ambiguity.

> I think the best implementation would be to move the code over as-is in a single commit, then apply the v2-specific changes on top in a separate commit.

I'm not sure what's meant by this - in this case would it be to move the code over then apply  the `content-type` change on top? I can probably do that, for what its worth (revert it from v1, move from v1 -> v2, apply it on v2)

Reply:
Okay, for now I have done just that.

Reply:
That's not really what I meant here; I thought that we had more changes on top of the current v1 code to make it "v2-compatible", but the issue is really that we already published v1 with the "wrong" feed API and we don't want people to use it.
I will rework the PR to avoid the double-revert.

Reply:
Thanks! I'm now rebased on top of this